server/runtime/src/main/java/org/infinispan/server/configuration/ServerConfigurationSerializer.java

```diff
@@ -278,6 +279,7 @@ private void writeEndpoints(ConfigurationWriter writer, EndpointsConfiguration e
       writer.writeStartElement(Element.ENDPOINTS);
       for (EndpointConfiguration endpoint : endpoints.endpoints()) {
          writer.writeStartElement(Element.ENDPOINT);
+         endpoint.singlePortRouter().attributes().write(writer);
```

Comment:
Without this here, the serialized version was missing the `<endpoint` attributes.

Something like:

```xml
<endpoint socket-binding="default" security-realm="default" io-threads="12" idle-timeout="1"
                   receive-buffer-size="2" send-buffer-size="3" tcp-keepalive="true"
                   tcp-nodelay="true">
```

Was serialized as:

```xml
<endpoint security-realm="default">
```